src/core/glob_matcher.cc

```diff
@@ -24,11 +24,26 @@ int stringmatchlen(const char* pattern, int patternLen, const char* string, int
         if (patternLen == 1)
           return 1; /* match */
         while (stringLen) {
-          if (stringmatchlen(pattern + 1, patternLen - 1, string, stringLen, nocase))
+          if (stringmatchlen_impl(pattern + 1, patternLen - 1, string, stringLen, nocase,
+                                  skipLongerMatches))
             return 1; /* match */
+          if (*skipLongerMatches)
+            return 0; /* no match */
           string++;
           stringLen--;
         }
+
+        /* There was no match for the rest of the pattern starting
+         * from anywhere in the rest of the string. If there were
+         * any '*' earlier in the pattern, we can terminate the
+         * search early without trying to match them to longer
+         * substrings. This is because a longer match for the
+         * earlier part of the pattern would require the rest of the
+         * pattern to match starting later in the string, and we
+         * have just determined that there is no match for the rest
+         * of the pattern starting from anywhere in the current
+         * string. */
+        *skipLongerMatches = 1;
```

Comment:
Why is this used as a `bool` but is marked as `int` and passed as `int *` ?

Reply:
it's taken from valkey codebase.